### ISSUE TITLE
Release v0.10.5

### ISSUE THREAD

**Issue by Skarlso** [external-secrets/external-secrets#4036]
This Issue tracks the next ESO release. Please follow the guideline below. If anything is missing or unclear, please add a comment to this issue so this can be improved after the release.

#### Preparation Tasks

- [ ] ask in `#external-secrets-dev` if we're ready for a release cut-off or if something needs to get urgently in
- [ ] docs: [stability & support page](https://external-secrets.io/main/introduction/stability-support/) is up to date
  - [ ] version table
  - [ ] Provider Stability and Support table
  - [ ] Provider Feature Support table
- [ ] docs: update [roadmap page](https://external-secrets.io/main/contributing/roadmap/)
- [ ] tidy up [Project Board](https://github.com/orgs/external-secrets/projects/2)
  - [ ] move issues to next milestone
  - [ ] close milestone

#### Release Execution

- [ ] Follow the [Release Process guide](https://external-secrets.io/main/contributing/release/)

#### After Release Tasks

- [ ] Announce release on `#external-secrets` in Slack

**Comment by Skarlso** [comment]
Largely done. Only OLM remains.

**Comment by faust64** [comment]
Hey @Skarlso 

Rather than log a separate issue ... We just got a new CSV showing, releasing external-secrets-operator v0.10.5
Somehow, those are pulling images from some ci.external-secrets.io registry, which is not allowed out, in our proxies.
Is this a mistake?
Previous images were hosted on ghcr.io, as far as I can tell.
Did it move? Should I request for my proxies to be reconfigured?

Thanks!

**Comment by Skarlso** [comment]
@gusfcarvalho

**Comment by gusfcarvalho** [comment]
@faust64 - ever since 0.10.4 this is the default behavior https://github.com/external-secrets/external-secrets/releases/tag/v0.10.4 see the release note here.

what do you mean by CSV?

**Comment by faust64** [comment]
A CSV or ClusterServiceVersion, represents the installation of a specific version, of an operator, through that Operator Lifecycle Manager
I have 60+ clusters here.

As far as I'm concerned, it's the first time I see this.
But yes, maybe you forgot to release 0.10.4 in OLM. As My CSV confirms, we'"re upgrading to 0.10.5 ... from 0.10.3. In all my 60+ clusters
```
external-secrets-operator.v0.10.5         External Secrets Operator                        0.10.5      external-secrets-operator.v0.10.3                  Succeeded
```

Word of notice though, when you want to introdue such changes, it would be nice you consider introducing a new channel.
With that OLM stuff, patches upgrades are installed with asking.
If there are changes, like registry, that requires me to file a case with my networking team ... I'm up for a good month waiting, with those deployments stuck. That's not cool. Major changes like those we expect to find in a changelog ... And shipping in a separate channel, requiring a conscious switch.
No more ESO until 2025 for me, I guess, ... great ...

From your 0.10.4 changelog:
> . While GHCR.io will keep on working for the foreseeable future, this change is to allow an eventual migration away from GHCR.

Nothing about that in 0.10.5 changelog ...
Wo ... what is it? Are those images still available in GHCR? Is there any way for me to get that operator back up in a timely manner?
Or should I start explaining my management that, well, maybe, ... ESO wasn't such a great idea after all ...

**Comment by Skarlso** [comment]
<img width="436" alt="Screenshot 2024-11-02 at 17 46 52" src="https://github.com/user-attachments/assets/f0e637a0-28ab-44c5-a20b-0392b901ed2e">

0.10.4 is also merged. Not sure why your deployments skipped it.

>Nothing about that in 0.10.5 changelog ...

Why would we repeat it?

You can / should still be able to pull from ghcr.io. Just add an override. They won't be deprecated any time soon.

The only things that changed are the helm chart values. Just overwrite 
```
oci.external-secrets.io
```
with `ghcr.io` for the webhook and the image repository and you're golden.

**Comment by Skarlso** [comment]
>Or should I start explaining my management that, well, maybe, ... ESO wasn't such a great idea after all ...

Please don't be petty. This is neither the forum nor the time for that.

**Comment by knelasevero** [comment]
InstallPlan Approval Mode: You can set the InstallPlan approval mode to Manual instead of Automatic.

It sounds like you are unsatisfied with how OLM is updating minor or patch or build updates automatically for you, you can change that globally for everything managed under OLM or just locally with Calatalog Source or Update Strategies.

The change was announced and we didn't intend on creating another channel since from the open source limited engineering time point of view the releases are already complex enough. 

https://externalsecrets.com/ was launched recently and there you can use ESO with extra support, different update strategies, dedicated release channels, more scrutiny regarding security updates and fips-compliant images, and any other custom requirements if you become a customer.

**Comment by gusfcarvalho** [comment]
> With that OLM stuff, patches upgrades are installed WITHOUT ASKING -- do you realize this? ... I mean, clusters all over sudden, all install the latest patches ... pretty useful for security. But here, I can guarantee you: I do not have time to read a changenog, nor would I have the means to stop that upgrade. Patch upgrades are just this ... "patches".

First of all, the issue happened due to your setup - all of our e2e tests are passing and we can confirm for most of our users this is not a breaking change, so I’d really like you to take that into account before blaming this should’ve been a major change instead.

Second of all, our official release channel is, and will always be, helm charts. Bundle manifests, rancher, OLM, everything else, are maintained with best effort. 

Our OLM release itself is based on a very old, deprecated, open source project - it is not and will not be our focus to maintain it (!). However, if you need to have it because you use it in production - **CONTRIBUTE 😁**. We accept both code/automation contributions and maintainership contributions. I’d be more than happy to let people that actually understand open shift to do that release as opposed to myself.

Now, I’d really appreciate if you stop trying to patronize for a free stuff that didn’t work the way you intended. It just demotivates us on keep maintaining a contribution from 2 years ago from someone who thought “having eso on OLM would be cool”.

maybe that’s it and we should just really stop supporting, heh? 🙄🙄

**Comment by faust64** [comment]
> Why would we repeat it?
I'm sorry, going back to yur changelogs. There is nothing about images no longer published to GHCR.
> GHCR.io will keep on working for the foreseeable future, this change is to allow an eventual migration away from GHCR.

It's not about repeating anything.
It's about actually communicating that, well, no: I can't pull from GHCR anymore, I absolutely have to migrate.
You do understand the concept of "will keep working for the foreseeable future", right?
Yet today, it doesn't?
What changed?

This is not at all how that works. Please try to familiarize yourself with OLM concepts. Channels in particular.
I'm fully aware of the auto-approval method, which is the default, and recommended running OpenShift, RedHat itself. When it comes to RedHat supported operators, I can assure you, this never happend.


Now ... And I'm sorry ... But as I'm unblocking this in a first lab ... 0.10.5 is also completely broken when it comes to pushsecrets.
Trying to change .type from existing secrets, which are immutable...


Maybe you're right.
Maybe I'll just stop deploying ESO ...
I used to have my own controllers dealing with this. They used to work very fine ... My management pushed to get OpenShift native crap ... and now here we are.

Good luck with your stuff ...

Those are patches upgrade, we do not expect to have patch upgrades changing the overall architecture of a deployment ... They should be ... patches, bug or security fixes.
Yes, maybe you should stop publishing to OLM, if you can't understand this.

**Comment by Skarlso** [comment]
Bye 👋

**Comment by knelasevero** [comment]
From Gustavo:
> However, if you need to have e.g. specific channels, etc. because you use it in production - CONTRIBUTE 😁

Also from Gustavo:
> Second of all, our official release channel is, and will always be, helm charts. Bundle manifests, rancher, OLM, everything else, are maintained with best effort.

> 0.10.5 is also completely broken when it comes to pushsecrets.
> Trying to change .type from existing secrets, which are Immutable 

Please report the issue you are seeing with push secrets in a separate issue.

---

If Redhat is now considering ESO as an productized "supported" operator in their internal Operator Hub, they should be helping us maintain this release method, and you should be creating support tickets in ServiceNow directed at them. 

Then again, if you want dedicated support, with all the guarantees, for free, I welcome you to familiarize yourself with the concept of "open source entitlement complex" in posts that you can find in HackerNews, Reddit, blogs and talks all over the internet. 

If you want to be treated like a customer, the relationship you are looking for won't be found here, on GitHub issues or open source forums. People here are working for free on their free time most of the time. And it's very difficult to cater to all the expectations of users and organisations if they don't help with contributions or supporting companies vendoring the project.
